experimental/apis/data/v0alpha1/query_test.go

```diff
@@ -90,6 +90,28 @@ func TestParseQueriesIntoQueryDataRequest(t *testing.T) {
 	})
 }
 
+func TestLegacyDataSourceRef(t *testing.T) {
+	type testWrapper struct {
+		Ref DataSourceRef `json:"ref"`
+	}
+
+	wrap := &testWrapper{}
+	err := json.Unmarshal([]byte(`{ "ref": {"type":"ttt", "uid":"UID"}}`), wrap)
+	require.NoError(t, err)
+	require.Equal(t, "ttt", wrap.Ref.Type)
+	require.Equal(t, "UID", wrap.Ref.UID)
+
+	err = json.Unmarshal([]byte(`{ "ref": "name"}`), wrap)
+	require.NoError(t, err)
+	require.Equal(t, "", wrap.Ref.Type)
+	require.Equal(t, "name", wrap.Ref.UID)
+
+	ref := &DataSourceRef{}
+	err = json.Unmarshal([]byte(`"aaa"`), ref) // string as reference
+	require.NoError(t, err)
+	require.Equal(t, "aaa", ref.UID)
```

Comment:
So ds name is moved into Ref.uid? No longer possible to know if there's a name or a uid I guess. Do we need to be able to distinguish/handle this legacy case? Can we fail instead getting this legacy format? Hard to know where these legacy forms comes from, imagine it should be migrated in dashboards?

Reply:
Yes they *should* be migrated!  but what do we do when they are not?  Currently the logic to look up by UID will fallback to lookup by name, so this works OK.  

Currently dashboard frontend always does this migration, the place we see issues is in very old alert definitions.